Enhanced error message if user imports another user's iNat observation

JoeCohen opened the pull request:
See [Pull Request Overview](https://github.com/MushroomObserver/mushroom-observer/pull/3371#pullrequestreview-3352092642) for details

The motivation for this PR is: one user says he cannot import an iNat observations because he always receives the error message "You cannot import an iNat observation created by someone else."
I cannot replicate the problem on my local (when logged in as that user) -- the import just works.
I hope that the enhanced message will help us to figure out what's going on. 

# Suggested Manual Test
## NOTE: requires knowing the iNat username of someone else who is not a superimporter

- Login to iNat as yourself
- Login to MO as someone else who is not a superimporter
- Import from iNat, filling in the form with:
  - that person's `inat_username` 
  - the iNat ID of someone else's observation
  - Submit
  
### Expected result:

> Status: Done
Imported: 0 of 1 observations
...
Errors: <your inat_username> is logged into iNat. To import iNat observations of <other person's inat_username> make sure that <other person's inat_username> is logged into iNat on your device
<img width="732" height="215" alt="Screenshot 2025-10-18 at 5 20 02 PM" src="https://github.com/user-attachments/assets/a85c7704-676d-458b-b17e-aca5f375027e" />

JoeCohen committed:
Enhance error msg when wrong user logged into iNat

JoeCohen committed:
Remove quotes in error message

It ovecomplicates the code

JoeCohen commented on config/locales/en.txt:
```diff
@@ -3106,7 +3106,7 @@
   inat_import_started: Your import has started. Refresh page to update results.
   inat_no_authorization: iNat import aborted; MO did not received authorization to access iNat user data
-  inat_wrong_user: You cannot import an iNat observation created by someone else. To import your own observation, make sure that noone else is logged into iNat on your device and that you entered the correct iNat username.
+  inat_wrong_user: "[me_user] is logged into iNat. To import iNat observations of [inat_username] make sure that [inat_username] is logged into iNat on your device"
```
Wrong. wrapping quotes needed here.

JoeCohen committed:
assert_includes instead of assert_match

responds to [CoPilot suggestion](https://github.com/MushroomObserver/mushroom-observer/pull/3371#discussion_r2441255191)

mo-nathan reviewed (approved):
I don't love the variable name `me_user`, but it's fine.  Feel free to merge.

JoeCohen committed:
Better variable name

Responds to [@mo-nathan comment](https://github.com/MushroomObserver/mushroom-observer/pull/3371#pullrequestreview-3354484828)
